src/selectedtests/task_mappings/mappings.py

```diff
@@ -62,57 +64,66 @@ def create_task_mappings(
+            try:
+                base_repo = _get_evg_project_and_init_repo(evg_api, evergreen_project, temp_dir)
+            except ValueError:
+                LOGGER.warning("Unexpected exception", exc_info=True)
+                raise
+
+            jobs = []
+            with Executor(max_workers=MAX_WORKERS) as exe:
+                for next_version, version, prev_version in windowed_iter(project_versions, 3):
+                    if not branch or not repo_name:
+                        branch = version.branch
+                        repo_name = version.repo
+
+                    LOGGER.info(f"Processing mappings for version", version=version.version_id)
+
                     try:
-                        base_repo = _get_evg_project_and_init_repo(
-                            evg_api, evergreen_project, temp_dir
+                        diff = _get_diff(base_repo, version.revision, prev_version.revision)
+                    except ValueError:
+                        LOGGER.warning("Unexpected exception", exc_info=True)
+                        continue
+
+                    changed_files = _get_filtered_files(diff, file_regex)
+
+                    if module_name:
+                        cur_module = _get_associated_module(version, module_name)
+                        prev_module = _get_associated_module(prev_version, module_name)
+                        if cur_module is not None and module_repo is None:
+                            module_repo = init_repo(
+                                temp_dir, cur_module.repo, cur_module.branch, cur_module.owner
+                            )
+
+                        module_changed_files = _get_module_changed_files(
+                            module_repo, cur_module, prev_module, module_file_regex
                         )
-                    except ValueError as e:
-                        LOGGER.exception(str(e))
-                        raise e
-                    branch = version.branch
-                    repo_name = version.repo
-
-                LOGGER.info(f"Processing mappings for version {version.version_id}")
-
-                try:
-                    diff = _get_diff(base_repo, version.revision, prev_version.revision)
-                except ValueError as e:
-                    print(e)
-                    continue
-
-                changed_files = _get_filtered_files(diff, file_regex)
-
-                if module_name is not None and module_name != "":
-                    cur_module = _get_associated_module(version, module_name)
-                    prev_module = _get_associated_module(prev_version, module_name)
-                    if cur_module is not None and module_repo is None:
-                        module_repo = init_repo(
-                            temp_dir, cur_module.repo, cur_module.branch, cur_module.owner
+                        changed_files = changed_files.union(module_changed_files)
+
+                    jobs.append(
```

Comment:
Here is where we are creating the futures. `exe.submit` adds a jobs to the executors list of jobs to do. The first parameters is the function to execute and the rest of the parameters just get passed to that function when it gets called. It returns a future that we track in the `jobs` list.